geode-management/src/main/java/org/apache/geode/management/internal/ClientClusterManagementService.java

```diff
@@ -194,9 +197,32 @@ private void checkIsRestful(CacheElement config) {
     }
   }
 
+  private <T extends ClusterManagementResult> T assertSuccessful(T result) {
+    if (!result.isSuccessful()) {
+      throw new RemoteClusterManagementException(result);
+    }
+    return result;
+  }
+
   public boolean isConnected() {
-    return restTemplate.getForEntity(RestfulEndpoint.URI_VERSION + "/ping", String.class)
-        .getBody().equals("pong");
+    try {
+      return restTemplate.getForEntity(RestfulEndpoint.URI_VERSION + "/ping", String.class)
+          .getBody().equals("pong");
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  /**
+   * not public because user code should *not* try to catch this exception by name, since other
+   * implementations of {@link ClusterManagementService} will throw different subtypes of
+   * RuntimeException
+   */
+  private static class RemoteClusterManagementException extends RuntimeException {
```

Comment:
ClusterManagementService interface should declare that all methods throw RuntimeException.  And LocatorClusterManagementService should also have checks to ensure no unsuccessful result is ever returned.

Reply:
fixed